### PR TITLE
[17.0][MIG] website_sale_product_pack

### PR DESCRIPTION
**Pull request by augusto-weiss**

### CODE REVIEW

**Comment by augusto-weiss** [comment]
@lef-adhoc @ioans73 @NICO-SOLUTIONS @pedrobaeza @bruno-zanotti 
Hello everyone!! Colud review this PR ??
Thanks all!!

**Review by bruno-zanotti — APPROVED** [review]
technical review LGTM

**Comment by NICO-SOLUTIONS** [comment]
@augusto-weiss @bruno-zanotti  
looks quite good. except for one issue... Just a question of what should be the result.

When in define a Product-Pack with Pack Type "Detailed" & Component Price "Detailed per component".... Right now it displays in the webshop the price of the pack without the component prices on the product overiew. When you add this pack to the cart, it adds the component prices to the price of the pack. 
From my point of view it should be handled in the following way. When there is a product with Pack Type "Detailed" & Component Price "Detailed per component", it should show the sum of the comonents without adding the price of the pack to it in the product page. When i add the pack to the cart, the line with the pack should be zero (or maybe a note) and the prices of the components should be listed. 
So the behaviour would be closer to the "totalized in main product" scenario. 
The advantage of "Detailed per component" is, that you even can add products with different tax rates. This is not possible with "totalized in main product". 
So its about getting rid of the pack price itself for "Detailed per component" packs.

We should consider this here: #173

What do you think? If there is another purpose for this kind of pack, please let me know.

**Comment by bruno-zanotti** [comment]
@NICO-SOLUTIONS 

Good point, that problem is also in older versions, so we decided to migrate the module as it is. Anyway, we made a PR in v17 with a pack's prices refactor that fixes lots of things included that issue, you can check it here https://github.com/OCA/product-pack/pull/173 (the fix is in the last commit)

**Review by NICO-SOLUTIONS — APPROVED** [review]
Technical review. LGTM

**Comment by pedrobaeza** [comment]
/ocabot migration website_sale_product_pack

NOTE: On v18, probably product combos will be enough and this module family won't be needed.

**Comment by NICO-SOLUTIONS** [comment]
> /ocabot migration website_sale_product_pack
> 
> NOTE: On v18, probably product combos will be enough and this module family won't be needed.

@pedrobaeza with combo choices you only can add each product with quantity = 1 to the choices. So i see the advantage of this module, where you can define the number of components. Havent check it in detail but for me this seem to be the bottelneck. Or am i missing something?

File: website_sale_product_pack/models/product_product.py [hunk 1]
```diff
@@ -17,8 +17,9 @@ def check_website_published(self):
             if unpublished:
                 raise ValidationError(
                     _(
-                        "You can't add unpublished products (%(unpublished_products)s) to"
-                        "a published pack (%(pack_name)s)"
+                        "You can't add unpublished products "
+                        "(%(unpublished_products)s)"
+                        "to a published pack (%(pack_name)s)"
```
Comment (pilarvargas-tecnativa) [inline website_sale_product_pack/models/product_product.py:22]: Is this a pre-commit change?

**Review by pilarvargas-tecnativa — APPROVED** [review]

**Pull request merged**